lib/sqlalchemy/util/langhelpers.py

```diff
@@ -292,17 +295,13 @@ def decorate(fn: _Fn) -> _Fn:
                 % metadata
             )
 
-        mod = sys.modules[fn.__module__]
-        env.update(vars(mod))
```

Comment:
this change seems unrelated

Reply:
it also breaks everything in the ci so it definitely needs to be reverted

Reply:
This is critical part.
Without this change, each decorated function has own large global dict.

I modified to copy defaults and kwdefaults too. We do not need compile it from source.

Reply:
I guess because the eval is against fake `__defaults__` we no longer need global env.   well that would be a good change